docker/docker-compose.1804.50.yaml

```diff
@@ -17,39 +17,6 @@ services:
 
   test:
     image: swift-nio:18.04-5.0
-    environment:
-      - MAX_ALLOCS_ALLOWED_1000_addHandlers=47050
```

Comment:
these alloc changes (only in 5.0 & 5.1 which I don't actually care about) are probably the result of the subclassing I guess?

Reply:
5.0 only I think, 5.1 could have stayed, it actually passed there, but as you didn't care about it, I thought it might go too.

Reply:
It seems that there is something with the subclasses that pushed the 5.0 to generate slightly more mallocs in some cases, but not in 5.1 and later. Perhaps some missing optimisation in 5.0.

Reply:
yeah, that's fine by me although I'd like to consciously make the decision that we're giving up some performance here. We should probably have a look at the code because not only do we have subclasses, we also have a generic subclass which can break specialisation which could be a real issue.

Reply:
@weissi do you want me to add the malloc checks back for 5.0/5.1?

Reply:
I don't mind having them or not. Said that, I think allocation changes can highlight performance changes even if they only change for some Swift versions. Given that we expect precisely 0 perf change with this PR, I think verifying that the old ones still work could be useful. But I'm happy without them too.

Reply:
They’re restored now.